main.tf

```diff
@@ -1,7 +1,7 @@
 terraform {
   required_version = ">= 0.12.6"
   required_providers {
-    azurerm = "~> 1.36.0"
+    azurerm = "~> 1.38.0"
```

Comment:
Why not use 1.40.0 since this is the latest version ?

Reply:
It will use 1.40 if it's available. This requires minimum v1.38 and makes it usable for users running v1.38-v1.40 (and newer)